app/models/manageiq/providers/ansible_tower/shared/automation_manager/tower_api.rb

```diff
@@ -9,16 +9,35 @@ def create_in_provider(manager_id, params)
 
       refresh(manager)
       find_by!(:manager_id => manager.id, :manager_ref => tower_object.id)
+    rescue AnsibleTowerClient::ClientError, ActiveRecord::RecordNotFound => error
+      raise
+    ensure
+      notify('create_in_provider', manager.id, params, error.nil?)
     end
 
     def create_in_provider_queue(manager_id, params)
       manager = ExtManagementSystem.find(manager_id)
-      action = "Creating #{name} with name=#{params[:name] || params['name']}"
+      action = "Creating #{name} with name=#{params[:name]}"
       queue(manager.my_zone, nil, "create_in_provider", [manager_id, params], action)
     end
 
     private
 
+    def notify(op, manager_id, params, success)
+      notify_params = params
```

Comment:
I was about to comment on unnecessary assignment until I read further.  Can you improve the readability here?  I think the code in the block is complicated enough to justify its own method.